### DISCUSSION TITLE
What's the intended way to "link" multiple "apps" together?

### DISCUSSION THREAD

**Discussion by danielrosehill** [Budibase/budibase#14057] [Help]
Let's say that you're using Budibase to build something like an internal "intranet" style platform.

As I use it, I can see that the intended functionality (I think!) is to have only one "function" per "app." Or at least given the fact that it's one app per URL, the idea is to create a number of apps rather than attempting to stuff everything into one of them.

If that's the case, I might very easily end up wanting to receive (and grant) access to multiple apps: One app for managing a directory table, another for managing another one, etc. 

Perhaps in large organisations it's more common for teams to require just one app to do their function, but in a small organisation one wears many hats!

My question is basically what's the intended way to create a cohesive user experience? Is the idea that we'd embed the individual app pages into a final platform? Much as data resources aren't shared between apps, navigations don't interlink either.  Or is the app toggle intended to serve as the internal menu of sorts?

Just curious (and confused) as to what's the intended way to link it all together.

**Comment by ConorWebb96** [comment]
Hey @danielrosehill,

There are a few ways.

1. You could looking into the [public API](https://docs.budibase.com/reference) with this you can get information from tables you have in other apps. This is more
2. You could use an external data source e.g. MySQL, Postgres etc. and then use that in another app as well. Other apps can be used to change parts of different databases or completely different flows for different users if you wish.

If you are using the internal Budibase data source you won't be able to do point 2.

Overall its definitely doable. I suppose the question for you is how you want to do it.

I hope this helps.

**Comment by danielrosehill** [comment]
Nice one, Conor!

I have a few Budibase "tenancies" set up now - one for personal stuff,
another for a passion project, and one, now, for work. I have Postgres
integrated with the first two and it works great. For the latter, I'm using
the built in DB as we don't have a managed database.

The point of friction I'm thinking about really is trying to get my
coworkers (less technical) to start using the internal app that I've built
(in all likelihood it's going to come out to something like 10 "apps" once
all the functionalities are adding in). It's tricky enough getting them to
use one service much less asking them to navigate through 10 URLs 😂

I'm thinking about something along the lines of embedding each app into a
Wordpress site and layering in a meta navigation that way. Although it
seems a bit counter intuitive.

Have you seen any good implementations that I might model?

Regards,
---------
*Daniel Rosehill*
Homepage <https://danielrosehill.com>

**Reply by ConorWebb96** [reply]
Hey @danielrosehill,

The best course of action then would be to use the first point I suggested. You have one master app that houses the details for all the other apps. Then in the other apps you'd setup API end points that allow the users to see/edit/delete/create from these apps. (depending on what you want specific apps to do of course.)

Alternatively you could just create one big app, and separate everything using permissions ([RBAC](https://docs.budibase.com/docs/user-roles#app-specific-roles)). You can create custom roles that can be used to prevent using from seeing certain screens, buttons components etc.

> I'm thinking about something along the lines of embedding each app into a Wordpress site and layering in a meta navigation that way. Although it seems a bit counter intuitive.

I'm not 100% on this, I believe they would still need to login to Budibase unless the screens are completely public. You could likely build a login page. Theres a great tutorial on how to do this [here](https://github.com/Budibase/budibase/discussions/6098).

**Comment by danielrosehill** [comment]
Cool!

Thanks for the help and thoughts, Conor.

I'm away from the office for a week or so, but looking into the API
integration will be high up my to-do list when I'm back.

If I ever figure it out and it might be of any interest I'd be happy to
pass on what worked (the office will certainly upgrade to one of the paid
tiers - I just need to get people to accept my invites first!).

Regards,
---------
*Daniel Rosehill*
Homepage <https://danielrosehill.com>
